### PR TITLE
Standalone rti

### PR DESCRIPTION
**Pull request by Soroosh129**
This branch implements a standalone RTI that can be installed universally on a UNIX-based system. 

The code for the RTI is located in the `org.lflang/src/lib/core/federated/RTI` folder. Perhaps, higher level directory (e.g., `util`) can be a better home for the RTI code.

### CODE REVIEW

File: org.lflang/src/lib/core/federated/net_common.h [hunk 1]
```diff
@@ -592,6 +600,35 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+
+/**
+ * A message that informs the RTI about connections between this federate and
+ * other federates where messages are routed through the RTI. Currently, this
+ * only includes logical connections when the coordination is centralized. This
+ * information is needed for the RTI to perform the centralized coordination.
+ * 
+ * @note Only information about the immidate neighbors is required. The RTI can
+ * transitively obtain the structure of the federation based on each federate's
+ * immidiate neighbor information.
```
Comment (edwardalee) [inline org.lflang/src/lib/core/federated/net_common.h:612]: immediate is misspelled.

**Review by edwardalee — APPROVED** [review]
Looks good!  Go for the merge...

File: org.lflang/src/lib/core/federated/net_common.h [hunk 2]
```diff
@@ -24,9 +25,10 @@ PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR B
 INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
 STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
+*/
+/**
  * @section DESCRIPTION
- * Header file for the runtime infrastructure for distributed Lingua Franca programs.
+ * Header file for common message types and definitions for distributed Lingua Franca programs.
```
Comment (lhstrh) [inline org.lflang/src/lib/core/federated/net_common.h:31]:
```suggestion
 * Header file for common message types and definitions for federated Lingua Franca programs.
```

**Review by lhstrh — APPROVED** [review]
This really cleans things up nicely. Great job!

**Pull request merged**